### PR TITLE
ForkedChainRef.forkchoice: Skip newBase calculation and skip chain finalization if finalizedHash is zero

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: nimbus/core/chain/forked_chain.nim [hunk 1]
```diff
@@ -428,6 +449,12 @@ proc forkChoice*(c: ForkedChainRef,
   # If there are multiple heads, find which chain headHash belongs to
   let head = ?c.findCanonicalHead(headHash)
```
Comment (advaita-saha) [inline nimbus/core/chain/forked_chain.nim:450]: when traced the error is originating from this line `Block hash is not part of any active chain`

Comment (advaita-saha) [reply nimbus/core/chain/forked_chain.nim:450]: And the same issue is still persisting, when engineAPI is ran on this branch

Comment (jangko) [reply nimbus/core/chain/forked_chain.nim:450]: hmm, it's weird. If the payload have been imported successfully by `engine_newPayload`, it should not error anymore. I will investigate deeper

Comment (jangko) [reply nimbus/core/chain/forked_chain.nim:454]: I've located the bug with my new written test case, can you please test it again @advaita-saha?

Comment (tersec) [reply nimbus/core/chain/forked_chain.nim:454]: > with my newly written test case

@jangko is this test case possible to include in this PR?

Comment (jangko) [reply nimbus/core/chain/forked_chain.nim:454]: yes

Comment (jangko) [reply nimbus/core/chain/forked_chain.nim:455]: https://github.com/status-im/nimbus-eth1/pull/2654/commits/c71659588cbcf98be9e31ab16c32af7f76e2f00a should fix the `eth_getBlockNumber` bug.

https://github.com/status-im/nimbus-eth1/pull/2654/commits/d0117282a01d5d3387993ab146b0606211185dbb is the test suite for engine api executed in ci.

**Comment by jangko** [comment]
Flushing finalized blocks to db will come in separate PR.

**Comment by advaita-saha** [comment]
> Flushing finalized blocks to db will come in separate PR.

I think it is getting flushed, the finalized blocks, just some issue with the head ( it stays at zero, with finalized moving forward ). Might be useful to know this before the db changes 
```bash
INF 2024-09-26 18:21:39.416+05:30 Database initialized                       base="(3F9F5FF29C0AED1F9522D74B955200E5DC7B30C9D7F5260CF9F264A1B9C9FC82, 6450)" finalized="(3F9F5FF29C0AED1F9522D74B955200E5DC7B30C9D7F5260CF9F264A1B9C9FC82, 6450)" head="(B5F7F912443C940F21FD611F12828D75B534364ED9E95CA4E307729A4661BDE4, 0)"
```

**Comment by advaita-saha** [comment]
@jangko also I think it would be nice to have some trace logs in the engine `api_forkchoice`, for future debugging purposes

**Pull request merged**